easybuild/easyconfigs/k/KofamScan_data/KofamScan_data-20230401-gompi-2021b.eb

```diff
@@ -0,0 +1,50 @@
+easyblock = 'Binary'
+
+name = 'KofamScan_data'
+version = '20230401'
+
+homepage = 'https://www.genome.jp/tools/kofamkoala/'
+description = """KofamKOALA assigns K numbers to the user's sequence data by 
+HMMER/HMMSEARCH against KOfam (a customized HMM database of KEGG Orthologs 
+(KOs)).  K number assignments with scores above the predefined thresholds 
+for individual KOs are more reliable than other proposed assignments.  Such 
+high score assignments are highlighted with asterisks '*' in the output.  
+The K number assignments facilitate the interpretation of the annotation 
+results by linking the user's sequence data to the KEGG pathways and EC 
+numbers.  [data only]."""
+# software_license = 'MIT'
+
+toolchain = {'name': 'gompi', 'version': '2021b'}
+
+source_urls = [
+    'https://www.genome.jp/ftp/db/kofam/',  # add archives/YYYY-MM-DD/ if moved
+]
+sources = [
+    'ko_list.gz',
+    'profiles.tar.gz',
+]
+checksums = [
+    '099a62576b1754f31d3182e54d8ea310b61a2cacd3f6b3d5762be51ab0c711dd',  # ko_list.gz
+    'e017ead8fd52677d0a20558b9fd504876558e1723e77ec6be79aa72e6dee90a2',  # profiles.tar.gz
+]
+
+dependencies = [
+    ('KofamScan',  '1.3.0'),  # hard coded latest program version
+]
+
+extract_sources = False
+
+install_cmd = 'mkdir -p %(installdir)s/db/ && '
+install_cmd += 'tar xfz profiles.tar.gz --directory %(installdir)s/db/ && '
+install_cmd += 'gzip -dc ko_list.gz > %(installdir)s/db/ko_list && '
+install_cmd += 'rm -f profiles.tar.gz && '
+install_cmd += 'rm -f ko_list.gz && '
+install_cmd += 'rm -rf $EBROOTKOFAMSCAN/db && '
+install_cmd += 'ln -s %(installdir)s/db $EBROOTKOFAMSCAN/db'
```

Comment:
oh, hm, so this modifies another installation? 

This seems backwards to me. Rather than symlinking stuff in here, of KofamScan depended on KofamScan_data, it could just write it's own configuration file to look for the files in this module instead, removing this need for a software to modify another installation (which will break in several setups, EB can write protect installation directories to ensure one doesn't do things like this)

Reply:
This is mimicking InterProScan_data, and I think doing this (symlink to program install dir) for KofamScan is rather reasonable than InterProScan. It seems to me that database files and program files are bundled together for InterProScan so we used to see one set of database files and program files under a directory of one version. For KofamScan, database files are monthly updated so it is possible to install a new version of database for the same program.

I am not very sure what to do to fit easybuild's standard. Do you think let KofamScan (the program module) depending on KofamScan_data and symlink a db dir (under KofamScan and by KofamScan.eb) to the db dir of KofamScan_data is acceptable?

Reply:
Thank you for the comment! So how about let's just delete KofamScan_data and add some comment and modloadmsg to ask people who installed KofamScan.eb to do some necessary modification?

Reply:
I would remove
```
install_cmd += 'echo "profile: %(installdir)s/db/profiles" > %(installdir)s/config.yml && '
install_cmd += 'echo "ko_list: %(installdir)s/db/ko_list" >> %(installdir)s/config.yml && '
```
from the software installation, forcing users to specify this themselves explicitly. 

and either do one of:
1. Don't provide the database at all (users just get the version they want for their research into their project area like many other things)
2. Provide Kofam-Data at SYSTEM level, add a modloadmsg to tell users how to use it; something along the lines of `"use specify kofam_scan --profile $EBROOTKOFAM_DATA/profile_stuff --ko-list $EBROOTKOFAM_DATA/ko-stuff ..."`

Since the database seems to be completely independent, and I suspect we don't want to keep up with PRs for this dataset every month, I'm leaning towards option 1, and leave that to the users themselves. 

I don't think we need any modloadmsg in this central repo, since there hasn't been any release yet with KofamScan included, we aren't breaking anything backwards compatibility. If you already provide this at your center and like to move it closer to upstream behavior, you could consider adding it locally.

Reply:
I did corresponding changes, removed KofamScan_data from this PR, and added some comments in KofamScan EB file. I kept modloadmsg but with some minor modification to remind users the command to enter and the example config file.